MiniApp/Classes/MiniAppLister.swift

```diff
@@ -1,2 +1,54 @@
 internal class MiniAppLister {
+    var environment: Environment
+    var miniAppClient: MiniAppClient
+
+    init(environment: Environment) {
+        self.environment = environment
+        self.miniAppClient = MiniAppClient()
+    }
+
+    func fetchList(completionHandler: @escaping (Result<[MiniAppInfo], Error>) -> Void) {
+        guard let url = environment.listingUrl else {
+            return completionHandler(.failure(self.invalidURLError()))
+        }
+
+        self.miniAppClient.requestFromServer(request: listingURLRequest(url: url)) { (result) in
+            switch result {
+            case .success(let responseData):
+                guard let decodeResponse = self.decodeListingResponse(with: responseData.data) as? [MiniAppInfo] else {
+                    return completionHandler(.failure(self.invalidResponseData()))
+                }
+                return completionHandler(.success(decodeResponse))
+            case .failure(let error):
+                return completionHandler(.failure(error))
+            }
+        }
+    }
+
+    func listingURLRequest(url: URL) -> URLRequest {
```

Comment:
I kind of would like to see more separation between the API request code and the business logic parts of the code. 

Right now, this isn't a big problem because the business logic for getting the mini app list is very simple. But with something more complicated like `MiniAppDownloader`, there will be a lot more business logic so I don't think we should be mixing the API code with it. Also, there could be more complicated logic required in the future for `MiniAppLister`.

On Android, all of the API code is in a separate `api` package: https://github.com/rakutentech/android-miniapp/tree/master/miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api
And then `MiniAppLister` is very simple: https://github.com/rakutentech/android-miniapp/blob/master/miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppLister.kt

So maybe most of the code that's currently in `MiniAppLister` could be moved to a separate `ListRequest` class and then `MiniAppLister` can just handle converting the response to a list of `MiniAppInfo`? What do you think?

Reply:
I think that's a good approach to have them separated like API request and business logic. 
I was thinking to create some separate class to handle API request. Will work on this new approach in new PR